docs: add charm configuration deprecations and updates for ops (#285)
canonical/anbox-cloud-docs: howto/upgrade/upgrade-anbox.md

    juju run --wait=1m lxd/0 clear-notification

If the LXD charm is deployed on a machine with an NVIDIA GPU installed, by default, the NVIDIA drivers are held from being upgraded in case of downtime for all running instances due to either a manual upgrade or an [unattended-upgrade](https://wiki.debian.org/UnattendedUpgrades). The downside to this is that the machine may miss security updates for the NVIDIA drivers. To manually upgrade the NVIDIA drivers, you need to run the following Juju action:

    juju run --wait=30m lxd/0 upgrade-gpu-drivers
</details>

```{important}
For users running LXD clusters with the LXD snap tracking a channel which is different than 5.21/stable, it is important that you set the charm configuration item `channel` *explicitly* to the currently running channel before running the `upgrade-cluster` command. This can be done using:

>     juju config lxd channel=5.21/stable

Reviewer: As mentioned above, can we add a bit more context about it E.g. `If the current LXD cluster consists of the LXD snap tracking from 5.0 channel, it is important that you set.... This can be done using:
    juju config lxd channel=5.0/stable
`

>     juju config lxd channel=5.21/stable

Reviewer: I would expect that the similar changes you made to the scale-up/scale-down section can also be applied to the upgrade-anbox section.

Not doing this might lead to a broken LXD cluster.
```

The first step to upgrade LXD is to update the corresponding charm to the latest revision using:

    juju refresh --channel=1.25/stable lxd

Updating the charm does not update the LXD snap or any of the internal packages on the LXD node. After updating the charm, check which nodes have package updates available using:

    juju run lxd/0 upgrade-info

If you have multiple Juju units for LXD nodes, make a note of all the Juju units having available updates.

Before proceeding make sure that there are no more active instances on the nodes. Then mark the LXD node as unschedulable within AMS which would make sure that no new instances are spawned on the LXD node. This can be done using:

>     juju run ams/leader start-maintenance node=”<node_name>”

Reviewer: Did we rename the action to `start-maintenance` without the leading `node-`?
Same below

Author: Thanks for spotting. Will fix this.

    juju run --wait=1m lxd/0 clear-notification

If the LXD charm is deployed on a machine with an NVIDIA GPU installed, by default, the NVIDIA drivers are held from being upgraded in case of downtime for all running instances due to either a manual upgrade or an [unattended-upgrade](https://wiki.debian.org/UnattendedUpgrades). The downside to this is that the machine may miss security updates for the NVIDIA drivers. To manually upgrade the NVIDIA drivers, you need to run the following Juju action:

    juju run --wait=30m lxd/0 upgrade-gpu-drivers
</details>

The first step to upgrade LXD is to update the charm itself to the latest revision using:

    juju refresh --channel=1.25/stable lxd

Updating the charm does not include updating the LXD snap or any of the internal packages on the LXD node. After updating the charm, check which nodes have package updates available using:

>     juju run lxd/0 upgrade-info

Reviewer 2: Doesn't this only tell us the info for a single node? Shouldn't we better run `juju run -a lxd upgrade-info` to get the info for all nodes with one call?

Author: Afaik there is no -a for `juju run` it is only for `juju exec`

> If you have multiple juju units for LXD nodes, make a note of all the juju units having available updates.

Reviewer 2: ```suggestion
If you have multiple Juju units for LXD nodes, make a note of all the Juju units having available updates.
```

> Before proceeding make sure that all the traffic to nodes that have updates available have been drained. Run:

Reviewer 2: ```suggestion
Before proceeding make sure that there are no more active instances on the nodes. Run:
```

    juju run ams/leader start-maintenance node=”<node_name>”

> where `<node_name>` refers to the LXD node name stored within AMS. This action marks the LXD node as unschedulable within AMS which would make sure that no new instances are spawned on the LXD node.

Reviewer 2: ```suggestion
where `<node_name>` refers to the LXD node name stored within AMS, e.g. `lxd0` if the Juju unit is `lxd/0`. This action marks the LXD node as unschedulable within AMS which would make sure that no new instances are spawned on the LXD node.
```

> After the node is successfully marked, run:

Reviewer 2: We need to explain what the following command does before we show the actual command I would say.

Reviewer 3: +1

    juju run lxd/<unit_number> --wait=30m upgrade-machine reboot=<true_or_false>

where `<unit_number>` refers to the Juju unit number on each node having available updates.

> This action will update the gpu drivers as well as installed kernel modules which might result in a reboot being required to successfully load the new modules and drivers. This can be done automatically by setting `reboot=true` or manually at a later point in time by setting `reboot=false` and then running `juju exec lxd/<unit_number> -- sudo reboot` when required.

Reviewer 2: ```suggestion
This action will update the GPU drivers as well as installed kernel modules which might result in a reboot being required to successfully load the new modules and drivers. This can be done automatically by setting `reboot=true` or manually at a later point in time by setting `reboot=false` and then running `juju exec lxd/<unit_number> -- sudo reboot` when required.
```

> After the updates to all the nodes finish, update the LXD snap on the node using:

Reviewer 2: ```suggestion
After the updates to all the nodes have finished, update the LXD snap on the node using:
```

    juju run lxd/<unit_number> upgrade-cluster.

```{important}

> For customers running LXD clusters with the LXD Snap tracking a channel which is different than 5.21/stable, please set the charm configuration item `channel` *explicitly* to the currently running channel before running the `upgrade-cluster` command. This can be done using:

Reviewer 2: ```suggestion
For users running LXD clusters with the LXD snap tracking a channel which is different than 5.21/stable, please set the charm configuration item `channel` *explicitly* to the currently used channel before running the `upgrade-cluster` command. This can be done using:
```

Reviewer 3: See my comment above that includes these changes

    juju config lxd channel=5.21/stable
Not doing this might lead to a broken LXD cluster.

> ```

Reviewer 2: I think we need to move the note above higher to increase visibility and awareness.

Reviewer 3: +1
